pinot-core/src/test/java/org/apache/pinot/queries/NativeAndLuceneComparisonTest.java

```diff
@@ -0,0 +1,295 @@
+      driver.init(config, recordReader);
+      driver.build();
+    }
+  }
+
+  private ImmutableSegment loadLuceneSegment()
+      throws Exception {
+    IndexLoadingConfig indexLoadingConfig = new IndexLoadingConfig();
+    Set<String> textIndexCols = new HashSet<>();
+    textIndexCols.add(DOMAIN_NAMES_COL_LUCENE);
+    indexLoadingConfig.setTextIndexColumns(textIndexCols);
+    Set<String> invertedIndexCols = new HashSet<>();
+    invertedIndexCols.add(DOMAIN_NAMES_COL_LUCENE);
+    indexLoadingConfig.setInvertedIndexColumns(invertedIndexCols);
+    return ImmutableSegmentLoader.load(new File(INDEX_DIR, SEGMENT_NAME_LUCENE), indexLoadingConfig);
+  }
+
+  private ImmutableSegment loadNativeIndexSegment()
+      throws Exception {
+    IndexLoadingConfig indexLoadingConfig = new IndexLoadingConfig();
+    Map<String, String> propertiesMap = new HashMap<>();
+    FSTType fstType = FSTType.NATIVE;
+    propertiesMap.put(FieldConfig.TEXT_FST_TYPE, FieldConfig.TEXT_NATIVE_FST_LITERAL);
+
+    Map<String, Map<String, String>> columnPropertiesParentMap = new HashMap<>();
+    Set<String> textIndexCols = new HashSet<>();
+    textIndexCols.add(DOMAIN_NAMES_COL_NATIVE);
+    indexLoadingConfig.setTextIndexColumns(textIndexCols);
+    indexLoadingConfig.setFSTIndexType(fstType);
+    Set<String> invertedIndexCols = new HashSet<>();
+    invertedIndexCols.add(DOMAIN_NAMES_COL_NATIVE);
+    indexLoadingConfig.setInvertedIndexColumns(invertedIndexCols);
+    columnPropertiesParentMap.put(DOMAIN_NAMES_COL_NATIVE, propertiesMap);
+    indexLoadingConfig.setColumnProperties(columnPropertiesParentMap);
+    return ImmutableSegmentLoader.load(new File(INDEX_DIR, SEGMENT_NAME_NATIVE), indexLoadingConfig);
+  }
+
+  private void testSelectionResults(String query, int expectedResultSize, List<Serializable[]> expectedResults) {
+    Operator<IntermediateResultsBlock> operator = getOperatorForSqlQuery(query);
+    IntermediateResultsBlock operatorResult = operator.nextBlock();
+    List<Object[]> resultset = (List<Object[]>) operatorResult.getSelectionResult();
+    Assert.assertNotNull(resultset);
+    Assert.assertEquals(resultset.size(), expectedResultSize);
+    if (expectedResults != null) {
+      for (int i = 0; i < expectedResultSize; i++) {
+        Object[] actualRow = resultset.get(i);
+        Object[] expectedRow = expectedResults.get(i);
+        Assert.assertEquals(actualRow.length, expectedRow.length);
+        for (int j = 0; j < actualRow.length; j++) {
+          Object actualColValue = actualRow[j];
+          Object expectedColValue = expectedRow[j];
+          Assert.assertEquals(actualColValue, expectedColValue);
+        }
+      }
+    }
+  }
+
+  @Test
+  public void testQueries() {
+    String query = "SELECT * FROM MyTable WHERE TEXT_MATCH(DOMAIN_NAMES_LUCENE, 'www.domain1%') LIMIT 50000";
+    _indexSegment = _luceneSegment;
+    _indexSegments = Arrays.asList(_indexSegment);
+    testSelectionResults(query, 256, null);
```

Comment:
We want to compare the result from lucene and native. Currently all the `expectedResults` are `null`